### PR TITLE
[us14533] Remove support for deprecated SAML provider addNew configur…

### PR DESCRIPTION
**Pull request by irinaepshteyn**
…ation property.

Signed-off-by: Matias matias.altman@ge.com

### CODE REVIEW

**Comment by irinaepshteyn** [comment]
Pull request build FAILED due to proxy issues - UnknownHostException.
http://sjc1jenkins08.crd.ge.com:8080/job/Predix-Security/job/uaa-build-n-test-pullrequest/39/console

File: model/src/main/java/org/cloudfoundry/identity/uaa/provider/SamlIdentityProviderDefinition.java [hunk 1]
```diff
@@ -242,11 +242,6 @@ public SamlIdentityProviderDefinition setZoneId(String zoneId) {
     }
 
     // For backward compatibility.
-    public boolean isAddNew() {
-        return isAddShadowUserOnLogin();
-    }
-
-    // For backward compatibility.
     public void setAddNew(boolean addNew) {
```
Comment (sanjeevchopra) [inline model/src/main/java/org/cloudfoundry/identity/uaa/provider/SamlIdentityProviderDefinition.java]: please remove the setter as well.

File: model/src/test/java/org/cloudfoundry/identity/uaa/provider/SamlIdentityProviderDefinitionTest.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                .setLinkText("Test SAML Provider");
+            samlDefinition.setAddNew(false);
+            samlDefinition.setEmailDomain(Arrays.asList("test.com", "test2.com"));
+        String config = JsonUtils.writeValueAsString(samlDefinition);
+        Assert.assertTrue(config.contains("\"addShadowUserOnLogin\":false"));
+        Assert.assertFalse(config.contains("addNew"));
+    }
+
+    @Test
+    public void test_that_json_containing_addNew_is_deserialized() {
```
Comment (sanjeevchopra) [inline model/src/test/java/org/cloudfoundry/identity/uaa/provider/SamlIdentityProviderDefinitionTest.java]: I was assuming that migration will ensure that there is no json with 'addNew' ?

File: server/src/test/java/org/cloudfoundry/identity/uaa/db/RemoveAddNewFromProviderConfig_V3_6_0_Tests.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+                new Integer(0));
+
+        // Verify that new property was assigned
+        assertFalse(((SamlIdentityProviderDefinition) this.provisioning
+                .retrieveByOrigin("depricatedPropertyOnlyIdp", zoneId).getConfig()).isAddShadowUserOnLogin());
+        assertTrue(((SamlIdentityProviderDefinition) this.provisioning.retrieveByOrigin("bothPropertiesIdp", zoneId)
+                .getConfig()).isAddShadowUserOnLogin());
+        assertTrue(((SamlIdentityProviderDefinition) this.provisioning.retrieveByOrigin("newPropertyOnlyIdp", zoneId)
+                .getConfig()).isAddShadowUserOnLogin());
+    }
```
Comment (sanjeevchopra) [inline server/src/test/java/org/cloudfoundry/identity/uaa/db/RemoveAddNewFromProviderConfig_V3_6_0_Tests.java]: 👍

File: server/src/test/java/org/cloudfoundry/identity/uaa/db/RemoveAddNewFromProviderConfig_V3_6_0_Tests.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        // with Config with addNew Flag only
+        String depricatedPropertyOnlyConfig = "{\"metaDataLocation\":\"mymetadata\",\"idpEntityAlias\":\"1D1DCE7769624FC0B8B8C11344EA67B8\","
+                + "\"zoneId\":\"apm\",\"nameID\":\"unspecified\",\"assertionConsumerIndex\":0,"
+                + "\"metadataTrustCheck\":false,\"showSamlLink\":false,\"socketFactoryClassName\":"
+                + "\"\",\"linkText\":\"Login with GE SSO staging\",\"iconUrl\":null,\"addNew\":false}";
+        this.provisionIdentityProvider(zoneId, "depricatedPropertyOnlyIdp", depricatedPropertyOnlyConfig);
+
+        // load Database
+        // with Config with addNew Flag and AddShadowUserOnLogin Flag
+        String bothPropertiesConfig = "{\"metaDataLocation\":\"mymetadata\",\"idpEntityAlias\":\"1D1DCE7769624FC0B8B8C11344EA67B8\","
```
Comment (sanjeevchopra) [inline server/src/test/java/org/cloudfoundry/identity/uaa/db/RemoveAddNewFromProviderConfig_V3_6_0_Tests.java]: not sure what the behavior should be when these conflict ?   @ilipkind ?

Comment (irinaepshteyn) [inline server/src/test/java/org/cloudfoundry/identity/uaa/db/RemoveAddNewFromProviderConfig_V3_6_0_Tests.java]: @sanjeevchopra, @ilipkind,

As was discussed previously with Ilya, if values are different, keep the value of addShadowUserOnLogin and remove addNew.  This and other requirements are captured in the user story [US14533](https://rally1.rallydev.com/#/58777512329d/detail/userstory/58795955045)

File: model/src/main/java/org/cloudfoundry/identity/uaa/provider/SamlIdentityProviderDefinition.java (hunk 1 above)
Comment (irinaepshteyn) [inline model/src/main/java/org/cloudfoundry/identity/uaa/provider/SamlIdentityProviderDefinition.java]: @sanjeevchopra,
Done

**Pull request closed without merging**